### DISCUSSION TITLE
can't get date-tag working in workflow

### DISCUSSION THREAD

**Discussion by pluim003** [community/community#37309] [Actions] [answered]
It was working with the old way (I just tried and tested) but then got a notification that it soon will be deprecated, so tried to modify it to the new way as described in: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/ 

But sofar without success. After approximately 15 tries I'm lost. How to solve this?

```name: Build and push latest Docker-Influxdb-Grafana-Telegraf image
on:
  # cron job to trigger the build on any push to dev
  push:
  #  branches:
  #    - 'dev'
  schedule:
  # cron job to trigger the build dialy (time in UTC)
    - cron:  '* 4 * * *'


jobs:
  docker:
    runs-on: ubuntu-latest
    steps:
      - name: Get tag name from date
        id: tag_date
        run: echo "TAG_DATE=$(date +'%Y%m%d')" >>GITHUB_ENV

      -
        name: Set up QEMU
        uses: docker/setup-qemu-action@v2
      -
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v2
      -
        name: Login to Docker Hub
        uses: docker/login-action@v2
        with:
          username: ${{ secrets.DOCKER_HUB_USERNAME }}
          password: ${{ secrets.DOCKER_HUB_TOKEN }}
      -
        name: Build and push latest
        uses: docker/build-push-action@v3
        with:
          platforms: linux/arm/v6,linux/arm/v7,linux/arm64/v8
          push: true
          tags: pluim003/influxdb-grafana-telegraf:latest, pluim003/influxdb-grafana-telegraf:${{ env.TAG_DATE }}
```

Error:
buildx failed with: ERROR: invalid tag "***/influxdb-grafana-telegraf:": invalid reference format so the env-variable seems not to be used.

**Comment by airtower-luna** [comment, marked as answer]
Your problem is in the redirection here:

```
        run: echo "TAG_DATE=$(date +'%Y%m%d')" >>GITHUB_ENV
```

`>>GITHUB_ENV` redirects to a file named `GITHUB_ENV`, but to set an environment variable you need to write to *the file named in the environment variable* `GITHUB_ENV`. So reference the environment variable instead:

```yaml
        run: echo "TAG_DATE=$(date +'%Y%m%d')" >>$GITHUB_ENV
```

**Reply by pluim003** [reply]
O wow, thanx. What a stupid mistake (from me). Must have type too fast and totallly overlooked it. Will give it a try later, as I now reverted back to the soon to be deprecated method.
